libs/atqr/domain/src/lib/valueObjects/notification.ts

```diff
@@ -0,0 +1,113 @@
+import { ValueObject } from '../../utils/valueObject';
+import { Guid } from '@tokilabs/lang';
+import { NotificationLogEntry } from './notificationLogEntry.valueObject';
+import { MessageTemplateId } from '../services';
+import {
+  NotificationChannel,
+  NotificationCategory,
+} from '../types/notification.types';
+export class Notification extends ValueObject<Notification> {
+  constructor(
+    public readonly id: Guid,
+    public readonly category: NotificationCategory,
+    public readonly templateId: MessageTemplateId,
+    public readonly allowedChannels: NotificationChannel[],
+    public readonly createdAt: Date,
+    public readonly requiresUserAction: boolean,
+    public readonly actionTaken: boolean,
+    public readonly contactLog: NotificationLogEntry[],
+    public readonly nextContactDate: Date,
+    /**
```

Comment:
Does this comment need to be here? (Note: do not use Portuguese)

Reply:
yes. it is needed . Saulo's order !

Reply:
"(campos que ela usa)" should have been replaced by a list of the fields used to calculate the property 🤣

Reply:
But these properties should be calculated and not received by the constructor